libraries/AP_RangeFinder/AP_RangeFinder_TOFSenseF_I2C.cpp

```diff
@@ -150,9 +150,6 @@ void AP_RangeFinder_TOFSenseF_I2C::update(void)
         state.distance_m = distance_mm * 0.001f;
         new_distance = false;
         update_status();
-    } else if (AP_HAL::millis() - state.last_reading_ms > 300) {
```

Comment:
I'm not sure some of the other drivers shouldn't more like this - i.e. removing the whole timeout handling in the driver and letting the frontend take care of what happens next.  Can't see why some places zero their fields and other places don't.

Reply:
i've gone through the others now and removed their timeout handling in most cases